app/views/postcodes/_postcode_lookup.html.haml

```diff
@@ -9,7 +9,7 @@
     = hidden_field_tag('constituency_id', @constituency.graph_id) if @constituency
     = hidden_field_tag('person_id', @person.graph_id) if @person
     = label_tag 'input', t('.search_by_postcode')
-    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'[0-9a-zA-Z ]{5,}', id:'input')
+    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([AZa-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z]))))\s?[0-9][A-Za-z]{2})$', id:'input')
```

Comment:
The Front-end pattern is only for basic validation i.e correct characters. The Back-end should validate the actual postcode using the comprehensive regex otherwise users will not receive the custom error message.

Previous pattern worked with spaces and three letter postcodes: https://regex101.com/r/CVvsRs/1